test/e2e/node_feature_discovery.go

```diff
@@ -0,0 +1,121 @@
+
+var _ = framework.KubeDescribe("NodeFeatureDiscovery", func() {
+	const (
+		labelPrefix       = "node.alpha.intel.com/nfd"
+		imageTag          = "e2e-tests"
+		fakeFeatureSource = "fake"
+	)
+	f := framework.NewDefaultFramework("node-feature-discovery")
+	var node *api.Node
+	var ns, name, image string
+	fakeFeatureLabels := map[string]string{
+		fmt.Sprintf("%s-%s-fakefeature1", labelPrefix, fakeFeatureSource): "true",
+		fmt.Sprintf("%s-%s-fakefeature2", labelPrefix, fakeFeatureSource): "true",
+		fmt.Sprintf("%s-%s-fakefeature3", labelPrefix, fakeFeatureSource): "true",
+	}
+
+	BeforeEach(func() {
+		ns = f.Namespace.Name
+		name = "node-feature-discovery-" + string(uuid.NewUUID())
+		image = fmt.Sprintf("quay.io/kubernetes_incubator/node-feature-discovery:%s", imageTag)
+		By("Selecting a non-master node")
+		if node == nil {
+			// Select the first non-master node.
+			nodes, err := f.ClientSet.Core().Nodes().List(api.ListOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			for _, n := range nodes.Items {
+				if !system.IsMasterNode(&n) {
+					node = &n
+					break
+				}
+			}
+		}
+		Expect(node).NotTo(BeNil())
+	})
+
+	It("Should decorate the selected node with the fake feature labels", func() {
+		By("Creating a node-feature-discovery pod on the selected node")
+		pod := &api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				Name: name,
+			},
+			Spec: api.PodSpec{
+				NodeName: node.Name,
+				Containers: []api.Container{
+					{
+						Name:    name,
+						Image:   image,
+						Command: []string{"/go/bin/node-feature-discovery", "--source=fake"},
+						Env: []api.EnvVar{
+							{
+								Name: "POD_NAME",
+								ValueFrom: &api.EnvVarSource{
+									FieldRef: &api.ObjectFieldSelector{
+										FieldPath: "metadata.name",
+									},
+								},
+							},
+							{
+								Name: "POD_NAMESPACE",
+								ValueFrom: &api.EnvVarSource{
+									FieldRef: &api.ObjectFieldSelector{
+										FieldPath: "metadata.namespace",
+									},
+								},
+							},
+						},
+					},
+				},
+				RestartPolicy: api.RestartPolicyNever,
+			},
+		}
+
+		_, err := f.ClientSet.Core().Pods(ns).Create(pod)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Waiting for the node-feature-discovery pod to succeed")
+		Expect(framework.WaitForPodSuccessInNamespace(f.ClientSet, name, ns))
+
+		By("Making sure the selected node was decorated with the fake feature labels")
+		options := api.ListOptions{LabelSelector: labels.SelectorFromSet(labels.Set(fakeFeatureLabels))}
+		matchedNodes, err := f.ClientSet.Core().Nodes().List(options)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(matchedNodes.Items[0].Name).To(Equal(node.Name))
+
+		By("Removing the fake feature and version labels advertised by the node-feature-discovery pod")
+		for key := range fakeFeatureLabels {
+			framework.RemoveLabelOffNode(f.ClientSet, node.Name, key)
+		}
+		framework.RemoveLabelOffNode(f.ClientSet, node.Name, "node.alpha.intel.com/node-feature-discovery.version")
+	})
```

Comment:
Here we could check that the node labels were actually cleaned... there's a function available already for it called [`VerifyLabelsRemoved`](https://github.com/balajismaniam/kubernetes/blob/e2e-tests-for-nfd/test/utils/density_utils.go#L93).